File: backend/src/main/java/com/safebite/backend/model/Noticia.java
package com.safebite.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "noticias")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Noticia {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String titulo;

    @Column(length = 500)
    private String resumen;

    @Column(length = 5000)
    private String contenido;

    private String imagenUrl;

    @ElementCollection(fetch = FetchType.EAGER)
    @CollectionTable(name = "noticia_etiquetas", joinColumns = @JoinColumn(name = "noticia_id"))
    @Column(name = "etiqueta")
    @Builder.Default
    private Set<String> etiquetas = new HashSet<>();

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "publicado_por")
    private Usuario publicadoPor;

    @Builder.Default
    private LocalDateTime fechaPublicacion = LocalDateTime.now();
}

File: frontend/lib/core/models/noticia.dart
class Noticia {
  final int id;
  final String titulo;
  final String? resumen;
  final String? contenido;
  final String? imagenUrl;
  final Set<String> etiquetas;
  final DateTime fechaPublicacion;

  Noticia({
    required this.id,
    required this.titulo,
    this.resumen,
    this.contenido,
    this.imagenUrl,
    required this.etiquetas,
    required this.fechaPublicacion,
  });

  factory Noticia.fromJson(Map<String, dynamic> json) => Noticia(
        id: json['id'],
        titulo: json['titulo'] ?? '',
        resumen: json['resumen'],
        contenido: json['contenido'],
        imagenUrl: json['imagenUrl'],
        etiquetas: ((json['etiquetas'] as List?) ?? []).map((e) => e.toString()).toSet(),
        fechaPublicacion:
            DateTime.tryParse(json['fechaPublicacion'] ?? '') ?? DateTime.now(),
      );

  Map<String, dynamic> toJson() => {
        'titulo': titulo,
        'resumen': resumen,
        'contenido': contenido,
        'imagenUrl': imagenUrl,
        'etiquetas': etiquetas.toList(),
      };
}
